server/src/test/java/org/elasticsearch/search/vectors/KnnByteVectorQueryBuilderTests.java

```diff
@@ -15,4 +15,13 @@ public class KnnByteVectorQueryBuilderTests extends AbstractKnnVectorQueryBuilde
     DenseVectorFieldMapper.ElementType elementType() {
         return DenseVectorFieldMapper.ElementType.BYTE;
     }
+
+    @Override
+    protected KnnVectorQueryBuilder createKnnVectorQueryBuilder(String fieldName, int numCands, Float similarity) {
+        byte[] vector = new byte[VECTOR_DIMENSION];
+        for (int i = 0; i < vector.length; i++) {
+            vector[i] = randomByte();
+        }
+        return new KnnVectorQueryBuilder(fieldName, vector, numCands, similarity);
```

Comment:
Tests relating to this are now failing with the following because we removed the "eager" parsing as byte, to also help distinguish between "true" byte (i.e. `hex`)  and "possibly" byte vectors.  

```
    java.lang.AssertionError: expected:<{
      "knn" : {
        "field" : "vector_alias",
        "query_vector" : [
          -27,
          -60,
          116
        ],
        "num_candidates" : 624,
        "similarity" : 0.56577635,
        "_name" : "some_name"
      }
    }> but was:<{
      "knn" : {
        "field" : "vector_alias",
        "query_vector" : [
          -27.0,
          -60.0,
          116.0
        ],
        "num_candidates" : 624,
        "similarity" : 0.56577635,
        "_name" : "some_name"
      }
```

We can add something to try and parse an incoming array as byte if possible, but I do wonder though if the false positives in this scenario, i.e. float vectors that happen to have only byte-accepted values, might hurt us in constantly transforming between the two, or if it would be better to just assume float from the very start and only convert to byte  if requested by `element_type`.  I think I am inclined towards the latter, as the transformation and checks will take place only when requested, with the caveat of `VectorData` being represented differently in older & new nodes (with the same end-result though).

Reply:
This happens because we call `ToXContent` which writes out an byte array and then FromXContent reads in the array as floats. 

@pmpailis is there a particular reason why you are not writing out `byte[]` as Hexadecimal strings? It seems like that is what the user is giving us, so we should write that out when we have a byte[] array. This way in parsing, we know its bytes.

Reply:
I was mainly worried of potential compatibility issues and whether this could be used in other places, but I guess we should be ok for `XContent` serialization.  (tbh as things stand, and without "eager" parsing, the only scenario where we would end up with a `byte[]` vector is through `hex` so this makes much sense).

Reply:
> I was mainly worried of potential compatibility issues and whether this could be used in other places

I am not sure if there will be any significant BWC issues, and the way around any such issues would be to just supply an array up front.

Reply:
While changing `toXContent` to write a hex-string instead of a byte array does solve the issue above, it seems  there were some BWC tests failing (`testBWCVersionSerializationQuery`). Will look into that and update accordingly.